File: java/src/main/java/org/hackerrank/DiagonalDifference.java
package org.hackerrank;

import java.util.List;

public class DiagonalDifference {
    public static int diagonalDifference(List<List<Integer>> arr) {
        int leftSum = 0;
        int rightSum = 0;
        int row = 0;
        for (int i = 0; i < arr.size(); i++) {
            int leftValue = arr.get(row).get(i);
            int rightValue = arr.get(row).get((arr.size() - 1) - i);
            leftSum += leftValue;
            rightSum += rightValue;
            row++;
        }
        return Math.abs(leftSum - rightSum);
    }
}

File: js/src/diagonalDifference.mjs
function diagonalDifference(arr) {
    // Write your code here
    let sumLeft = 0;
    let sumRight = 0;
    let row = 0;
    for(let i=0; i < arr.length; i++) {
        let leftValue = arr[row][i];
        let rightValue = arr[row][(arr.length - 1) - i];
        sumLeft += leftValue;
        sumRight += rightValue;
        row++;
    }
    return Math.abs(sumLeft - sumRight);
}

module.exports = diagonalDifference;
